msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -219,6 +220,14 @@ public void setComponent(final String component) {
     }
 
     /**
+     * To send extra parameters to token endpoint, which provides the possibility if developer wants to test features that have not been enabled on prod yet.
```

Comment:
this actually encourages external developers to use the property. Consider updating to "Custom query parameters which may be sent to the STS for dogfood testing. This parameter should not be set by the developers as it may have adverse effect on the application."